v2/delay/delay.go

```diff
@@ -196,7 +200,7 @@ func Func(key string, i interface{}) *Function {
 	}
 
 	if old := funcs[f.key]; old != nil {
-		old.err = fmt.Errorf("multiple functions registered for %s in %s", key, file)
+		old.err = fmt.Errorf("multiple functions registered for %s", key)
```

Comment:
I'm thinking that maybe we should just panic here.  Is there any reason we'd want to allow a program with this issue to initialize successfully?  If I were an application owner, I'd much rather my program fail to start, then to have it start up with a lurking time bomb.

Reply:
I'm fine with panic here, but I'm concerned about how this would change the current behavior of Func.

Reply:
Makes sense.  How about we leave `Func` with the old collision behavior, but panic here in `Register`.  If  you factor lines 175-200 into a private helper function, then you can reuse all of that logic in both places, and they can each implement this last bit differently.

Reply:
On second thought, I think we should just return error for both functions with the same key. The calls to these functions would fail but not panic, since I'm worried that panic here might crash user's apps, say when they do mass migration and such. The new added test here should clearly show the change in behavior.

Reply:
How about both - add a `MustRegister()` that panics on error?

Reply:
Hmm, that sounds reasonable. I trust your judgement so, code updated, PTAL 😃